service/src/test/java/com/TWCC/controller/EventControllerIntegrationTest.java

```diff
@@ -112,13 +139,171 @@ void testGetEventsByIdNotFound() {
         eventRepository.saveAll(events);
 
         // When
-        ResponseEntity<?> response = eventController.getEventsById(5);
+        ResponseEntity<?> response = eventController.getEventsById(500);
+
+        try {
+            // Then
+            assertEquals(HttpStatus.NOT_FOUND, response.getStatusCode());
+            MessageResponse messageResponse = (MessageResponse) response.getBody();
+            assertEquals("Event ID: 500 does not exist", messageResponse.getMessage());
+            assertEquals(HttpStatus.NOT_FOUND.value(), messageResponse.getStatus());
+        } catch (Exception e) {
+            e.printStackTrace();
+            assertTrue(false);
+        }
+    }
+
+    @Test
+    void testGetEventsByAddress() {
+        // Given
+        eventRepository.saveAll(events);
+
+        // When
+        List<Event> events = eventController.getEventsByAddress(event2.getAddress());
+
+        //Then
+        assertEquals(1, events.size());
+        assertEquals(event2.getName(), events.get(0).getName());
+    }
+
+    @Test
+    void testFilterEvent() {
+        // Given
+        eventRepository.saveAll(events);
+
+        // When
+        Map<String, String> allParams = new HashMap<String, String>();
+        allParams.put("address", "Columbia");
+        List<Event> filteredEvents = eventService.filterEvents(allParams, events);
+
+        assertEquals(2, filteredEvents.size());
+        assertEquals(allParams.get("address"), filteredEvents.get(0).getAddress());
+        assertEquals(allParams.get("address"), filteredEvents.get(1).getAddress());
+    }
+
+    @Test
+    void testCreateEventSuccessfully() {
+        // Given
+        Event newEvent = new Event(4, "NYU", 22,
+            "test session for course 4156",
+            "This is the test session for course 4156",
+            12.5, 122.34, 0, "www.google.com", 4,
+            "social, study",
+            new Timestamp(new Date().getTime() - 10),
+            new Timestamp(new Date().getTime() + 5),
+            new Timestamp(new Date().getTime() + 10)
+        );
+
+        TwccUser newUser = new TwccUser(
+            1,
+            "foo",
+            "bar",
+            25,
+            "foobar",
+            "12345",
+            "foobar@baz.com"
+        );
+
+        userController.registerUser(newUser);
+        ResponseEntity<?> loginResponse = userController.loginUser(new LoginRequest(newUser.getUsername(), "12345"));
+        JwtResponse jwtResponse = (JwtResponse) loginResponse.getBody();
+        System.out.println(jwtResponse.getToken());
+
+        // When
+        ResponseEntity<?> response = eventController.createEvent(newEvent, "Bearer " + jwtResponse.getToken());
```

Comment:
Could you rename this from response -> `eventResponse`. Actually I noticed this in all your tests. Might want to rename all of them and subsequently the variable names where you do response.getBody(). But no biggie.